https-dns-proxy: update to 2023.12.26-1

stangri opened the pull request:
Maintainer: me
Compile tested: x86_64, Dell EMC Edge620, OpenWrt 23.05.4
Run tested: x86_64, Dell EMC Edge620, OpenWrt 23.05.4

Description:

Makefile:
* update to latest upstream version
* remove PKG_SOURCE_DATE/PKG_SOURCE_RELEASE as they are no longer needed
* set TARGET_CFLAGS/TARGET_LDFLAGS
* update CMAKE_OPTIONS
* add CONFIGURE_ARGS to prepare for building with HTTP/3
* update package URL to upstream repo instead of documentation
* update package/description
* add README.md with link to documentation

init-script:
* do not run within image builder
* add a line which can be uncommented to remove outdated doh_server entries

020-src-options.c-add-version.patch:
* remove it, as it's no longer needed with version set in CMAKE_OPTIONS

BKPepe commented:
I am not sure if we need a readme that points from the upstream repository to the downstream repository or even to the downstream documentation. Because based on your own adding README to your packages, everyone can do that, and we would have so many unrelevant commits regarding README for every single package, which should be in your repository, not in ours. Let's keep the Git history clear and clean, please.

From my side, I am against having such a README like this. You should use OpenWrt documentation for it, which can be found here: https://openwrt.org/start . If we would look for README, which we do have for Python, it is completely different and relevant for build system.

Before merging this, there should be some conclusion about this, so it is suggested to wait for other maintainers.

stangri commented:
Actually, I'd rather other maintainers chime in on the more pressing issue of cherry-picked commit from snapshots not building for 23.05 due to mismatched hash: https://github.com/openwrt/packages/pull/24737#issuecomment-2267200028.

stangri commented:
> and we would have so many unrelevant commits regarding README for every single package, which should be in your repository, not in ours. Let's keep the Git history clear and clean, please.

Precisely! Having README in another repo allows:
* No spamming of neither upstream nor OpenWrt repo with PRs for when changes to README are needed
* Have different revisions of the README linked from different revisions of the package/luci app
* Persistence of the README if the package name is changed or package is deleted from either repo

The README included in this PR addresses all these issues by including a link to the README in a different repository.
